lib/mongoid/extensions/big_decimal.rb

```diff
@@ -58,7 +68,31 @@ def demongoize(object)
         #
         # @return [ String, nil ] A String representing the object or nil.
         def mongoize(object)
-          object && object.numeric? ? object.to_s : nil
+          if object
+            if Mongoid.map_big_decimal_to_decimal128
+              mongoize_to_decimal128(object)
+            elsif object.numeric?
+              object.to_s
+            end
+          end
+        end
+
+        private
+
+        def demongoize_from_decimal128(object)
+          if Mongoid.map_big_decimal_to_decimal128
+            object.to_big_decimal
+          else
+            raise Mongoid::Errors::UnmappedBSONType.new(object)
```

Comment:
I don't think I agree with this. When you opt in to using the BigDecimal type, you expect to lose some precision. If you really cared about trailing zeroes then you would have used Decimal128.

Reply:
If a user specified BigDecimal field type and database data is a Decimal128, I think they should get back a BigDecimal. This combination won't be currently produced by Mongoid from what I understand.

Reply:
done